### PR TITLE
Add Dokka documentation to artifacts

### PR DESCRIPTION
**Pull request by tripolkaandrey**
This PR adds Dokka-generated documentation as a publish artifact. 

It is done by using functionality introduced in [this PR in config](https://github.com/SpineEventEngine/config/pull/352).

Also, while updating the config, the dead code from buildSrc, like the `groovy` directory, was removed.

## Important details
- Breaking changes were introduced in the `2.11.0` version of `error_prone_core` leading to a compilation error. The `2.10.0` version was forced for all [other](https://mvnrepository.com/artifact/com.google.errorprone/error_prone_core/2.10.0) Error Prone parts to resolve the issue.

- Dokka has an issue when dealing with Metaspace. The problem and its workaround were mentioned in [this PR in config](https://github.com/SpineEventEngine/config/pull/352). This repository is large enough to run into `OutOfSpaceError`, so `MaxMetaspaceSize` was manually set as a parameter to the Gradle build CLI command in `build-on-ubuntu` and `build-on-windows` GitHub workflows to overcome the issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -219,20 +224,35 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-core</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:237]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -219,20 +224,35 @@ all modules and does not describe the project structure per-subproject.
     <artifactId>org.jacoco.ant</artifactId>
     <version>0.8.7</version>
   </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:227]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -219,20 +224,35 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:232]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:237]: @sonatype-lift ignore

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:232]: @sonatype-lift ignore

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:227]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:237]: @sonatype-lift ignore

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:227]: @sonatype-lift ignore

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:232]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:232]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:237]: @sonatype-lift ignore

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**